ray-operator/apis/ray/v1/rayservice_types.go

```diff
@@ -80,6 +80,12 @@ type RayServiceSpec struct {
 
 // RayServiceStatuses defines the observed state of RayService
 type RayServiceStatuses struct {
+	// Represents the latest available observations of a RayService's current state.
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=type
+	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
```

Comment:
May I ask what's the order for the conditions?

Reply:
What do you mean? There are only 2 conditions for `RayService`. `Ready` and `UpgradeInProgress`. https://github.com/ray-project/kuberay/blob/4a828dfd399027d34212b34a418233dd91e93684/ray-operator/apis/ray/v1/rayservice_types.go#L155-L160

Reply:
I want to know whether the service is in the requested status, a function like `IsServiceUnderStatus(status string)`.
Discussed offline, k8s go client already provides util functions to check.